tools/pipelines/templates/include-set-package-version.yml

```diff
@@ -145,9 +145,10 @@ steps:
     workingDirectory: ${{ parameters.buildDirectory }}
     filePath: $(Build.SourcesDirectory)/scripts/update-package-version.sh
 
-# This template only runs for non-PR main-branch builds, generating manifest files based on the current date and build number. 
-#  Enabling this template for every PR run risks overwriting existing manifest files uploaded to Azure blobs. Therefore, it's crucial to restrict this template to commits merged into the main branch.
-- ${{ if and(ne(variables['Build.Reason'], 'PullRequest'), eq(variables['Build.SourceBranch'], 'refs/heads/main')) }}: 
+# Only generate manifest files for runs in the internal project (i.e. CI runs, not PR run), for the main branch, and
+# for the build of the client release group (we don't need manifests for anything else).
+# Enabling this template for every PR run risks overwriting existing manifest files uploaded to Azure blobs. Therefore, it's crucial to restrict this template to commits merged into the main branch.
+- ${{ if and(eq(variables['System.TeamProject'], 'internal'), eq(variables['Build.SourceBranch'], 'refs/heads/main'), eq(parameters.tagName, 'client')) }}:
```

Comment:
Adding the `parameters.tagName == client` condition is the actual fix.

Also replaced the `Build.Reason != PullRequest` condition with `System.TeamProject == internal`, for better alignment with other places where we do similar checks. The internal project is configured to not do PR builds.